frontend/src/common/queries/collections.ts

```diff
@@ -37,12 +38,26 @@ export interface RevisionResponse extends CollectionResponse {
   revision: REVISION_STATUS;
 }
 
-async function fetchCollections(): Promise<CollectionResponse[]> {
+// if we return as a Map it will be easier to fetch collections by id
+export type CollectionResponsesMap = Map<
+  CollectionResponse["id"],
+  Map<VISIBILITY_TYPE, CollectionResponse>
+>;
+
+async function fetchCollections(): Promise<CollectionResponsesMap> {
   const json = await (
     await fetch(API_URL + API.COLLECTIONS, DEFAULT_FETCH_OPTIONS)
   ).json();
 
-  return json.collections;
+  const collectionsMap: CollectionResponsesMap = new Map();
+
+  for (const collection of json.collections as CollectionResponse[]) {
+    const collectionsWithId = collectionsMap.get(collection.id) || new Map();
+    collectionsWithId.set(collection.visibility, collection);
+    collectionsMap.set(collection.id, collectionsWithId);
+  }
+
+  return collectionsMap;
```

Comment:
Storing as `Map<CollectionResponse["id"], Map<VISIBILITY_TYPE, CollectionResponse>>` now for ease of access